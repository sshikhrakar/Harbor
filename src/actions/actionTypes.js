export const LOGIN_VIA_EMAIL = 'AUTH/LOGIN_VIA_EMAIL';
export const LOGIN_VIA_EMAIL_FULFILLED = 'AUTH/LOGIN_VIA_EMAIL_FULFILLED';
export const LOGIN_VIA_EMAIL_ERRORED = 'AUTH/LOGIN_VIA_EMAIL_ERRORED';
