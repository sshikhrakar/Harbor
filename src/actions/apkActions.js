import {
  INSTALL_APK,
  INSTALL_APK_ERRORED,
  INSTALL_APK_FULFILLED,
} from './actionTypes';

/**
 * Action to install a locally available apk.
 *
 * @param {String} path
 * @returns {Object}
 */
export function installApk(path) {
  return {
    type: INSTALL_APK,
    payload: {
      path,
    },
  };
}

/**
 * Action to indicate fulfillment of apk install.
 *
 * @returns {Object}
 */
export function installApkFulfilled() {
  return {
    type: INSTALL_APK_FULFILLED,
  };
}

/**
 * Action to indicate failure of apk install.
 *
 * @returns {Object}
 */
export function installApkErrored() {
  return {
    type: INSTALL_APK_ERRORED,
  };
}
