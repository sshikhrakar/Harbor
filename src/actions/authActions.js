import {
  LOGIN_VIA_EMAIL,
  LOGIN_VIA_EMAIL_ERRORED,
  LOGIN_VIA_EMAIL_FULFILLED,
} from './actionTypes';

/**
 * Login via Email action creator.
 *
 * @param {String} email
 * @param {String} password
 * @returns {Object}
 */
export function loginViaEmail(email, password) {
  return {
    type: LOGIN_VIA_EMAIL,
    payload: {
      email,
      password,
    },
  };
}

/**
 * Success dispatch for login.
 *
 * @param {Object} payload
 * @returns {Object}
 */
export function loginViaEmailFulfilled() {
  return {
    type: LOGIN_VIA_EMAIL_FULFILLED,
  };
}

/**
 * Error dispatch for login.
 *
 * @param {Object} payload
 * @returns {Object}
 */
export function loginViaEmailErrored() {
  return {
    type: LOGIN_VIA_EMAIL_ERRORED,
  };
}
