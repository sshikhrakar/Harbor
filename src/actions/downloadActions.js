import {
  DOWNLOAD_STARTED,
  DOWNLOAD_ERRORED,
  DOWNLOAD_COMPLETED,
  UPDATE_CURRENT_DOWNLOAD_PROGRESS,
} from '../actions/actionTypes';

/**
 * Start a download.
 *
 * @param {Object} project
 * @returns {Object}
 */
export function startDownload(project) {
  return {
    type: DOWNLOAD_STARTED,
    payload: {
      project,
    },
  };
}

/**
 * Stop a download.
 *
 * @param {Object} project
 * @returns {Object}
 */
export function completeDownload(project) {
  return {
    type: DOWNLOAD_COMPLETED,
    payload: {
      project,
    },
  };
}

/**
 * Download has errored.
 *
 * @param {Object} error
 * @returns {Object}
 */
export function downloadErrored(error) {
  return {
    type: DOWNLOAD_ERRORED,
    payload: {
      error,
    },
  };
}

/**
 * Update progress value for current download. Called repeatedly for progress bar update.
 *
 * @param {String | Number} progress
 * @returns {Object}
 */
export function updateCurrentDownloadProgress(progress) {
  return {
    type: UPDATE_CURRENT_DOWNLOAD_PROGRESS,
    payload: {
      progress,
    },
  };
}
