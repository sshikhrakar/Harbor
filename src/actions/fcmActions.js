import {
  REGISTER_FCM_TOKEN,
  REGISTER_FCM_TOKEN_FULFILLED,
  REGISTER_FCM_TOKEN_ERRORED,
  REFRESH_FCM_TOKEN,
  REFRESH_FCM_TOKEN_FULFILLED,
  REFRESH_FCM_TOKEN_ERRORED,
} from './actionTypes';

/**
 * Dispatch FCM token registration action.
 *
 * @returns {Object}
 */
export function registerFcmToken() {
  return {
    type: REGISTER_FCM_TOKEN,
  };
}


/**
 * Dispatch when FCM token is successfully registered.
 *
 * @returns {Object}
 */
export function registerFcmTokenFulfilled() {
  return {
    type: REGISTER_FCM_TOKEN_FULFILLED,
  };
}

/**
 * Dispatch when FCM token registration fails.
 *
 * @param {Object} err
 * @returns {Object}
 */
export function registerFcmTokenErrored(err) {
  return {
    type: REGISTER_FCM_TOKEN_ERRORED,
    payload: {
      err,
    },
  };
}


/**
 * Action creator for failed refresh of FCM token.
 *
 * @param {String} token
 * @returns {Object}
 */
export function refreshFcmToken(token) {
  return {
    type: REFRESH_FCM_TOKEN,
    payload: {
      token,
    },
  };
}

/**
 * Action creator for failed refresh of FCM token.
 *
 * @returns {Object}
 */
export function refreshFcmTokenErrored() {
  return {
    type: REFRESH_FCM_TOKEN_ERRORED,
  };
}

/**
 * Action creator for successful refresh of FCM token.
 *
 * @returns {Object}
 */
export function refreshFcmTokenFulfilled() {
  return {
    type: REFRESH_FCM_TOKEN_FULFILLED,
  };
}
