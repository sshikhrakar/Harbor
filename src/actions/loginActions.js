export const LOGIN_VIA_EMAIL = 'LOGIN_VIA_EMAIL';

/**
 * Login via Email action creator.
 *
 * @param {String} email
 * @param {String} password
 * @returns {Object}
 */
function loginViaEmail(email, password) {
  return {
    type: LOGIN_VIA_EMAIL,
    payload: {
      email,
      password,
    },
  };
}

export {
  loginViaEmail,
};
