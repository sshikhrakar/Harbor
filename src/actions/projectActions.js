import {
  FETCH_ALL_PROJECTS,
  FETCH_ALL_PROJECTS_ERRORED,
  FETCH_ALL_PROJECTS_FULFILLED,

  SET_SELECTED_PROJECT,
} from './actionTypes';

/**
 * Fetch all projects from server.
 *
 * @returns {Object}
 */
export function fetchAllProjects() {
  return {
    type: FETCH_ALL_PROJECTS,
  };
}

/**
 * Error in fetching of all projects.
 *
 * @param {Object} error
 * @returns {Object}
 */
export function fetchAllProjectsErrored(error) {
  return {
    type: FETCH_ALL_PROJECTS_ERRORED,
    error,
  };
}

/**
 * Successful fetch of all projects.
 *
 * @param {Object} projects
 * @returns {Object}
 */
export function fetchAllProjectsFulfilled(projects) {
  return {
    type: FETCH_ALL_PROJECTS_FULFILLED,
    payload: {
      projects,
    },
  };
}

/**
 * Set a project as target, with its primary key.
 *
 * @param {String} projectKey
 * @returns {Object}
 */
export function setSelectedProject(projectKey) {
  return {
    type: SET_SELECTED_PROJECT,
    payload: {
      project: projectKey,
    },
  };
}
