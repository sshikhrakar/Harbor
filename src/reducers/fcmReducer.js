import { REGISTER_FCM_TOKEN_FULFILLED } from '../actions/actionTypes';

export const initialState = {
  fcmTokenRegistered: false,
};

/**
 * Reducer for firebase FCM state.
 *
 * @param {Object} state=initialState
 * @param {Object} action
 * @returns {Object}
 */
function fcmReducer(state = initialState, action) {
  switch(action.type) {
    case REGISTER_FCM_TOKEN_FULFILLED:
      return {
        ...state,
        fcmTokenRegistered: true,
      };

    default:
      return state;
  }
}

export default fcmReducer;
