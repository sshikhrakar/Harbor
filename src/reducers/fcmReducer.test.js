import fcmReducer, { initialState } from './fcmReducer';
import { registerFcmTokenFulfilled } from '../actions/fcmActions';

describe('REDUCER: FCM', () => {

  it('should return initialState when state is undefined and action is empty', () => {
    expect(fcmReducer(undefined, {})).toEqual(initialState);
  });

  it('should receive registerFcmTokenFulfilled action and set state', () => {
    expect(fcmReducer(undefined, registerFcmTokenFulfilled())).toMatchSnapshot();
  });

});
