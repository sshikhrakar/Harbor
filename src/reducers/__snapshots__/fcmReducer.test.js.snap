// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REDUCER: FCM should receive registerFcmTokenFulfilled action and set state 1`] = `
Object {
  "fcmTokenRegistered": true,
}
`;
